dashboard-frontend/cypress/integration/listing-studies.spec.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  makeServer,
+  createStudyResource,
+  createStudyResources,
+} from '../../src/server';
+import { Response } from 'miragejs';
+
+describe('Given an authenticated user', () => {
+  let server: ReturnType<typeof makeServer>;
+
+  beforeEach(() => {
+    server = makeServer({ environment: 'test' });
+  });
+
+  afterEach(() => {
+    server.shutdown();
+  });
+
+  describe("When he hasn't created any Study and visits the home page", () => {
+    beforeEach(() => {
+      cy.visit('/');
+      assertLoaderAppears();
+    });
+
+    it('He sees information on how to create a Study', () => {
+      cy.contains('Contact info@vlab.digital and create your first Study.');
+      cy.contains('info@vlab.digital').should(
+        'have.attr',
+        'href',
+        'mailto:info@vlab.digital?subject=[New Account] Create first Study'
+      );
+    });
+  });
+
+  describe('When he has created one Study and visits the home page', () => {
+    beforeEach(() => {
+      createStudyResource(server, {
+        name: 'Study 1',
+        createdAt: new Date('Tue, 09 Mar 2021 01:39:09 GMT').getTime(),
+      });
+
+      cy.visit('/');
+
+      assertLoaderAppears();
+    });
+
+    it('He sees the name and creation date of the Study', () => {
+      cy.get('[data-testid="study-list-item"]')
+        .should('contain', 'Study 1')
+        .should('contain', 'Created on March 09, 2021');
+    });
+  });
+
+  describe('When he has created 20 Studies and visits the home page', () => {
+    beforeEach(() => {
+      createStudyResources(server, 20);
+
+      cy.visit('/');
+
+      assertLoaderAppears();
+    });
+
+    it('He sees a list of 10 studies', () => {
+      cy.get('[data-testid="study-list-item"]').should('have.length', 10);
+    });
+
+    it('He sees the 20 studies after paginating once', () => {
+      cy.scrollTo('bottom');
+      assertLoaderAppears();
+      cy.get('[data-testid="study-list-item"]').should('have.length', 20);
+    });
+  });
+
+  describe('When he visits the home page and there is an error while fetching the Studies', () => {
+    beforeEach(() => {
+      server.get('/studies', () => {
+        return new Response(
+          500,
+          { 'content-type': 'application/json' },
+          { error: ['Database cannot handle more connections'] }
+        );
+      });
+
+      cy.visit('/');
+
+      assertLoaderAppears();
+    });
+
+    it('He sees an error page with a button to retry fetching the Studies', () => {
+      cy.contains('Something went wrong while fetching the Studies.').then(
```

Comment:
This is great, but can we add the actual error message from the server? Even if they don't use it, it helps users to know if they're seeing the same message what to do.

Reply:
Yes definitely 👌  I'll display the error message from the server if available and fallback to the generic message in case none is provided or the error is a timeout.